src/httpc.h

```diff
@@ -133,11 +133,11 @@ struct httpc_request {
 	 */
 	bool set_keep_alive_header;
 	/**
-	 * True when Transfer-Encoding: chunked must be
-	 * set before execution automatically if chunked
-	 * io enabled.
+	 * It's the number of bytes of data in the body of the
+	 * request. The value is equal to "Content-Length"
+	 * header or 0 otherwise.
 	 */
```

Comment:
To think: a zero length request is a valid request. Should we always use the chunked encoding for it?

Reply:
Moreover: we use a signed type here (and `curl_off_t` is signed as well), so there is no problem to use -1 as the special value.

Reply:
You are right:

https://www.rfc-editor.org/rfc/rfc9110#field.content-length

Reply:
Updated:

1. `-1` by default for `req->content_length`.
2. Allow non-negative value for the `Content-Length` header.